evm/src/interfaces/ICircleIntegration.sol

```diff
@@ -31,59 +32,239 @@ interface ICircleIntegration {
         bytes payload;
     }
 
-    function transferTokensWithPayload(TransferParameters memory transferParams, uint32 batchId, bytes memory payload)
-        external
-        payable
-        returns (uint64 messageSequence);
+    /**
+     * @notice Parameters used for outbound transfers.
+     */
+    struct TransferParameters {
+        address token;
+        uint256 amount;
+        uint16 targetChain;
+        bytes32 mintRecipient;
+    }
 
-    function redeemTokensWithPayload(RedeemParameters memory params)
+    /**
+     * @notice Parameters used for inbound transfers.
+     */
+    struct RedeemParameters {
+        bytes encodedVaa;
+        bytes cctpMessage;
+        bytes cctpAttestation;
+    }
+
+    /**
+     * @notice Emitted when Circle-supported assets have been minted to the mintRecipient
+     * @param emitterChainId Wormhole chain ID of emitter contract on source chain
+     * @param emitterAddress Address (bytes32 zero-left-padded) of emitter on source chain
+     * @param sequence Sequence of Wormhole message used to mint tokens
+     */
+    event Redeemed(
+        uint16 indexed emitterChainId, bytes32 indexed emitterAddress, uint64 indexed sequence
+    );
+
+    /**
+     * @notice `transferTokensWithPayload` calls the Circle Bridge contract to burn Circle-supported
+     *  tokens. It emits a Wormhole message containing a user-specified payload with instructions for
+     *  what to do with the Circle-supported assets once they have been minted on the target chain.
+     * @dev Reverts if:
+     * - user passes insufficient value to pay Wormhole message fee
+     * - `token` is not supported by Circle Bridge
+     * - `amount` is zero
+     * - `targetChain` is not supported
+     * - `mintRecipient` is bytes32(0)
+     * @param transferParams Struct containing the following attributes:
+     * - `token` Address of the token to be burned
+     * - `amount` Amount of `token` to be burned
+     * - `targetChain` Wormhole chain ID of the target blockchain
+     * - `mintRecipient` The recipient wallet or contract address on the target chain
```

Comment:
Since we removed validation checks for a lot of these parameters, it might be nice to mention that the Circle contract will revert instead.

Reply:
Good call out. I need to go through everything to review and fix the comments

Reply:
Fixed comments here: https://github.com/wormhole-foundation/wormhole-circle-integration/pull/48/commits/3fdb1014a884922660f94a0e545a2c9ec846b870.